specs/phase0/validator.md

```diff
@@ -606,15 +608,28 @@ def get_aggregate_and_proof_signature(state: BeaconState,
 
 ## Phase 0 attestation subnet stability
 
-Because Phase 0 does not have shards and thus does not have Shard Committees, there is no stable backbone to the attestation subnets (`beacon_attestation_{subnet_id}`). To provide this stability, each validator must:
+Because Phase 0 does not have shards and thus does not have Shard Committees, there is no stable backbone to the attestation subnets (`beacon_attestation_{subnet_id}`). To provide this stability, each beacon node should:
 
-* Randomly select and remain subscribed to `RANDOM_SUBNETS_PER_VALIDATOR` attestation subnets
-* Maintain advertisement of the randomly selected subnets in their node's ENR `attnets` entry by setting the randomly selected `subnet_id` bits to `True` (e.g. `ENR["attnets"][subnet_id] = True`) for all persistent attestation subnets
-* Set the lifetime of each random subscription to a random number of epochs between `EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION` and `2 * EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION]`. At the end of life for a subscription, select a new random subnet, update subnet subscriptions, and publish an updated ENR
+* Remain subscribed to `SUBNETS_PER_NODE` for `EPOCHS_PER_SUBNET_SUBSCRIPTION` epochs.
+* Maintain advertisement of the selected subnets in their node's ENR `attnets` entry by setting the selected `subnet_id` bits to `True` (e.g. `ENR["attnets"][subnet_id] = True`) for all persistent attestation subnets.
+* Select these subnets based on their node-id as specified by the following `compute_subnets(node_id,epoch)` function.
 
-*Note*: Short lived beacon committee assignments should not be added in into the ENR `attnets` entry.
+```python
+def compute_subnet(node_id: int, epoch: Epoch, index: int) -> int:
+    node_id_prefix = node_id >> (256 - ATTESTATION_SUBNET_PREFIX_BITS)
+    permutation_seed = hash(uint_to_bytes(epoch // EPOCHS_PER_SUBNET_SUBSCRIPTION))
+    permutated_prefix = compute_shuffled_index(node_id_prefix, 1 << ATTESTATION_SUBNET_PREFIX_BITS, permutation_seed)
+    return (permutated_prefix + index) % ATTESTATION_SUBNET_COUNT
+```
+
+```python
+def compute_subnets(node_id: int, epoch: Epoch) -> Sequence[int]:
+    return [compute_subnet(node_id, epoch, idx) for idx in range(SUBNETS_PER_NODE)]
+```
+
+*Note*: Nodes should subscribe to new subnets and remain subscribed to old subnets for at least one epoch. Nodes should pick a random duration to unsubscribe from old subnets to smooth the transition on the exact epoch boundary of which the shuffling changes. 
```

Comment:
We should parameterize the random duration in the same fashion as was done for sync committee subnets here:
https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/validator.md#sync-committee-subnet-stability

We could use `ATTESTATION_SUBNET_COUNT` but that maybe excessive, so it could be a new param `ATTESTATION_SUBNET_RANDOM_COUNT` or something similar which could be set to a much smaller number .